### PR TITLE
fprintd-tod: fix the build

### PR DESCRIPTION
**Pull request by ymatsiuk**
###### Motivation for this change
#144627
#144843 
[#zhfmsk](https://nixhax.github.io/zhf-21.11/index.en.html)

Pin  `fprintd-tod` to `1.90.9` until [libfprint-tod-vfs0090](https://gitlab.freedesktop.org/3v1n0/libfprint-tod-vfs0090) supports newer version

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ymatsiuk** [comment]
Result of `nixpkgs-review pr 146737` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fprintd-tod</li>
  </ul>
</details>

**Review by balsoft — CHANGES_REQUESTED** [review]
Please move the fetch to the package. It doesn't belong in `top-level`.

Also, your override wouldn't work anyways,

**Comment by ymatsiuk** [comment]
Just to clarify: `fprintd-tod` is essentially another instance of `fprintd` but with replaced dependency. So what I wanted to do is to override `src` attribute.

I'm not sure if I understood the request correctly. Do I have to move the override into a separate file? (I see similar overrides in the same file btw, didn't know it's not allowed)

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -5307,9 +5307,17 @@ with pkgs;
 
   fprintd = callPackage ../tools/security/fprintd { };
 
-  fprintd-tod = callPackage ../tools/security/fprintd {
+  fprintd-tod = fprintd.overrideAttrs (oldAttrs: rec {
     libfprint = libfprint-tod;
```
Comment (balsoft) [inline pkgs/top-level/all-packages.nix:5311]: This doesn't work now that you use `overrideAttrs` instead of `override`.

**Comment by balsoft** [comment]
> I'm not sure if I understood the request correctly. Do I have to move the override into a separate file? (I see similar overrides in the same file btw, didn't know it's not allowed)

It's allowed but I think it's a bit better to put it in a separate file, since it's not really a temporary override but more of a separate package.

**Review by balsoft — APPROVED** [review]

**Pull request merged**